lib/hal/font_sasteroids2.h

```diff
@@ -3,6 +3,7 @@
 // www.svgalib.org site. Graciously provided
 // by John Remyn, originally distributed
 // with sasteroids.
+__attribute__((deprecated("deprecated as of 05/2021", "Use font_terminal.h or font_unscii_16.h instead")))
```

Comment:
This looks strange to me (the attribute being in the middle of a definition, instead of pre- or post-fix), are you sure it's syntactically correct?

Reply:
No I'm not sure. I especially noted that unclearness in the PR description. 
> As I wasn't sure how to deprecate a font, I just added the deprecated attribute to the top of the file font_sasteroids2.h

I hoped you might have a better idea :p

Reply:
Good catch! I missed this.

This is definitely incorrect as the preprocessor runs independently.
So if you did
```c
#include <hal/font_sasteroids2.h>
void shinyNewFunction() {}
```
it would deprecate your `shinyNewFunction`

---

Just add a `#warning` or we should skip deprecation and remove it directly.
- There'll only be a slight visual change when switching to the new font, so it's an acceptable trade-off.
- Nothing prevents users to copy the file into their own project if they require old behaviour.
- Does anybody even use this file? Especially with unclear legal state it's unlikely.
- Migrating to another file is as easy as renaming the include (which is less work, than us bikeshedding over it).

Reply:
Resolved by removing `font_sasteroids2.h`. Not sure if having multiple(or so many) fonts should be part of "hal" anyway, imo.